rclcpp/include/rclcpp/subscription.hpp

```diff
@@ -76,15 +76,15 @@ class SubscriptionBase
   get_topic_name() const;
 
   RCLCPP_PUBLIC
-  rcl_subscription_t *
+  std::shared_ptr<rcl_subscription_t>
   get_subscription_handle();
```

Comment:
These functions make up part of the public API for rclcpp - should we provide any versions that allow return type of rcl_subscription_t* to remain backwards compatible?

Reply:
I'll let OSRF decide :)